### ISSUE TITLE
Flow loss (grad_loss_fn) converge to zero

### ISSUE THREAD

**Issue by xkdlsxm2** [voxelmorph/voxelmorph#287]
**Task** (what are you trying to do/register?)

I am registering mono-modal MR liver images with different motion states to have same the motion state.
There are 5 motion states, and I try to register first 4 motion states to the last motion state. 
so, my moving image channel is 4, and 1 channel of fixed image. 

**What have you tried**

Since my input channel is 4, i have 12 channels of flow.
So, I changed forward part of the SpatialTransformer method like below
`
def forward(self, src_, flow_):
  
        warp = []

        for ch_idx in range(4):

            src = src_[:, ch_idx:ch_idx + 1, ...]

            flow = flow_[:, ch_idx * 3:ch_idx * 3 + 3, ...]

            new_locs = self.grid + flow

            shape = flow.shape[2:]

            # Need to normalize grid values to [-1, 1] for resampler
            for i in range(len(shape)):
                new_locs[:, i, ...] = 2 * (new_locs[:, i, ...] / (shape[i] - 1) - 0.5)

            if len(shape) == 2:
                new_locs = new_locs.permute(0, 2, 3, 1)
                new_locs = new_locs[..., [1, 0]]

            elif len(shape) == 3:
                new_locs = new_locs.permute(0, 2, 3, 4, 1)
                new_locs = new_locs[..., [2, 1, 0]]

            warp.append(nnf.grid_sample(src, new_locs, mode=self.mode))

        return torch.cat(warp, dim=1)
`

And instead of using MSE or CC, I used pytorch built-in method, "L1" since my another parallel experiment model was regulated by L1. 
I found grad_loss was too small with the parameter you mentioned in the paper (1 for MSE and 0.1 for CC), so I increased the regularizing parameter as 100 to 1000. 
for the first 3 iterations in the first epoch, grad_loss increases a lot, but after 3 epochs, grad_loss finally converges to zero. I mean not exactly zero, but it almost zero so the final result was not very different to the input. 

And, 3D image have 2 different size of z direction, 30 and 35, I declare two variables for SpatialTransformer like
`

        self.spatial_transform_35 = SpatialTransformer((35, *args.image_size))

        self.spatial_transform_30 = SpatialTransformer((30, *args.image_size))
`

in initializer of cvpr2018_net class, and in the forward method, I controlled the spatial_trasnform_xx by

`

        if src.shape[2] == 35:
            y = self.spatial_transform_35(src, flow)

        elif src.shape[2] == 30:
            y = self.spatial_transform_30(src, flow)
`

And since I need to use the same network to another parallel experiment, I plug in my ResUNet model in initializer of cvpr2018_net clase like

`

         if model == "UNet_voxelmorph":
            self.model = unet_core(dim=dim, feats=args.feats, in_channels=args.in_ch, out_channels=args.out_ch, full_size=full_size)

        elif model == "ResUNet_voxelmorph":
            self.model = ResUNet(in_channels=args.in_ch, out_channels=args.out_ch, n_features=args.feats, trilinear=True, kernel=args.kernel, pad=args.padding)
`

        
These three parts (4 channels of moving image, 2 different variable of spatial_transform_xx and plugged-in ResUNet model) are only things I changed from your model. 
**Details of experiments**

The model network is ResUNet which I implemented, and I used L1 loss and Adam optimier. Learning rate is 1e-2 to 1e-3 and it decays by 0.9 every 2 epochs. I run 30 epochs and each epoch iterates 116 which is the number of my 3D dataset. 
![7-Loss_plot](https://user-images.githubusercontent.com/12407801/113099807-d49f6700-91fa-11eb-9801-1827198d06b5.png)
This plot is loss plot and corresponding PSNR and SSIM
The training loss for the first 2 epochs are high because of high grad_loss, but after it converge to zero, then the loss almost converges at some point. 
As you can see in PSNR or SSIM, it increases very slightly, and it doesn't show any big difference in output image comparing to the input. 

This image is the result image.
You can ignore the first column ( I just input to compare with another parallel experiment), and the second column is average image of the 4 moving images, the third is the output of 1 channel, and the last column is the target. 

![7_explicit - b50_0](https://user-images.githubusercontent.com/12407801/113100056-2cd66900-91fb-11eb-85d9-95bb159f431b.png)

What would be the problem to update flow? or how to prevent converging grad_loss to zero?

Thank you very much 

Please carefully specify details about your experiments. If you are training, when what is the setup? What loss are you using? What does the convergence look like? If you are registering, please show example inputs and outputs. etc.
